runtime/pkg/duration/duration.go

```diff
@@ -104,3 +133,15 @@ func (d Duration) Sub(t time.Time) time.Time {
 	td := time.Duration(d.Second)*time.Second + time.Duration(d.Minute)*time.Minute + time.Duration(d.Hour)*time.Hour
 	return t.Add(-td)
 }
+
+type ToDateDuration struct {
+	anchor runtimev1.TimeGrain
+}
```

Comment:
Would propose changing it to `TruncToDate` instead, `ToDate` sounds like "convert to date".

Also, a small docstring would be helpful.